Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Payment Flow/PaymentFormViewController.m

```diff
@@ -0,0 +1,144 @@
+//  Copyright © 2020 Kristie Huang. All rights reserved.
+//
+
+#import "PaymentFormViewController.h"
+#import <Stripe/Stripe.h>
+#import "Utils.h"
+#import "APIManager.h"
+
+@interface PaymentFormViewController ()
+@property (weak) STPPaymentCardTextField *cardTextField;
+@property (weak) UIButton *payButton;
+@property (strong) NSString *paymentIntentClientSecret;
+
+@end
+
+@implementation PaymentFormViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    // Set up the Stripe card text field
+    STPPaymentCardTextField *cardTextField = [[STPPaymentCardTextField alloc] init];
+    self.cardTextField = cardTextField;
+    UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
+    button.layer.cornerRadius = 5;
+    button.backgroundColor = [UIColor systemBlueColor];
+    button.titleLabel.font = [UIFont systemFontOfSize:22];
+    [button setTitle:@"Pay" forState:UIControlStateNormal];
+    [button addTarget:self action:@selector(pay) forControlEvents:UIControlEventTouchUpInside];
+    self.payButton = button;
+    UIStackView *stackView = [[UIStackView alloc] initWithArrangedSubviews:@[cardTextField, button]];
+    stackView.axis = UILayoutConstraintAxisVertical;
+    stackView.translatesAutoresizingMaskIntoConstraints = NO;
+    stackView.spacing = 20;
+    [self.view addSubview:stackView];
+    [NSLayoutConstraint activateConstraints:@[
+        [stackView.leftAnchor constraintEqualToSystemSpacingAfterAnchor:self.view.leftAnchor multiplier:2],
+        [self.view.rightAnchor constraintEqualToSystemSpacingAfterAnchor:stackView.rightAnchor multiplier:2],
+        [stackView.topAnchor constraintEqualToSystemSpacingBelowAnchor:self.view.safeAreaLayoutGuide.topAnchor multiplier:2],
+    ]];
+    
+    [self createPaymentIntent];
+    
+}
+
+- (void)createPaymentIntent {
+    NSString *backendURL = [APIManager getAPISecretKeysDict][@"Backend_Server_Url"];
+    // Create a PaymentIntent by calling your server's endpoint.
+    NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:@"%@create-payment-intent", backendURL]];
+    NSDictionary *json = @{
+        @"items": @[
+                @{@"merchantId1": @"abc",
+                  @"price": @8000
+                },
+                @{@"merchantId2": @"cde",
+                  @"price": @8000
+                },
+                @{@"merchantId3": @"efg",
+                  @"price": @1000 //in cents
+                }
+        ],
+    };
+    NSData *body = [NSJSONSerialization dataWithJSONObject:json options:0 error:nil];
+    NSMutableURLRequest *request = [[NSURLRequest requestWithURL:url] mutableCopy];
+    [request setHTTPMethod:@"POST"];
+    [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
+    [request setHTTPBody:body];
+    NSURLSessionTask *task = [[NSURLSession sharedSession] dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *requestError) {
```

Comment:
Payments logic should be decoupled from the view controller, I'd recommend moving this logic to a new PaymentsAPIManager

Reply:
Will do in next commit !